### PR TITLE
Move Elasticsearch client config to elasticsearch/client package

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Move Elasticsearch client config to `elasticsearch/client` package. Also removed start server dependencies on dynamic config.

**Why?**
`config` package is splitted up to dedicated packages.

**How did you test it?**
Existing tests, local run.

**Potential risks**
No risks.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: common/config/persistence.go [hunk 1]
```diff
@@ -90,11 +95,7 @@ func (c *Persistence) validateAdvancedVisibility() error {
 		return fmt.Errorf("persistence config: advanced visibility datastore %q: missing config", c.AdvancedVisibilityStore)
 	}
 
-	if advancedVisibilityDataStore.ElasticSearch == nil {
-		return fmt.Errorf("persistence config: advanced visibility datastore %q: must provide config for elasticsearch store", c.AdvancedVisibilityStore)
-	}
-
-	if err := advancedVisibilityDataStore.ElasticSearch.validate(c.AdvancedVisibilityStore); err != nil {
+	if err := advancedVisibilityDataStore.Elasticsearch.Validate(c.AdvancedVisibilityStore); err != nil {
```
Comment (yiminc) [inline common/config/persistence.go:98]: can just do:
return advancedVisibilityDataStore.Elasticsearch.Validate(c.AdvancedVisibilityStore)

File: temporal/server.go [hunk 2]
```diff
@@ -386,44 +383,39 @@ func (s *Server) newBootstrapParams(
...
 
 	advancedVisibilityStore, ok := s.so.config.Persistence.DataStores[s.so.config.Persistence.AdvancedVisibilityStore]
 	if !ok {
-		return nil, nil, fmt.Errorf("unable to find advanced visibility store in config for %q key", s.so.config.Persistence.AdvancedVisibilityStore)
-	}
-
-	indexName := advancedVisibilityStore.ElasticSearch.GetVisibilityIndex()
-	if len(indexName) == 0 {
-		return nil, nil, errors.New("visibility index in missing in Elasticsearch config")
+		return nil, nil, nil
```
Comment (yiminc) [inline temporal/server.go:393]: shouldn't you return error here?

Comment (alexshtin) [reply temporal/server.go:393]: At this moment, config is already validated and this should never happened. If `AdvancedVisibilityConfigExist` (checked few lines above) it is valid and must present. But I agree to return error here.

File: common/config/persistence.go (hunk 1 above)
Comment (yiminc) [reply common/config/persistence.go:98]: I disagree, but it is ok. :)

**Review by yiminc — APPROVED** [review]

**Pull request merged**